src/test/mina_automation/utils.ml

```diff
@@ -30,3 +30,75 @@ let dedup_and_sort_archive_files files : string list =
 let force_kill process =
   Process.send_signal process Core.Signal.kill ;
   Deferred.map (Process.wait process) ~f:Or_error.return
+
+(** [get_memory_usage_mb pid] retrieves the memory usage in megabytes for the process
+  with the given [pid].
+  
+  This function reads the process memory information from [/proc/pid/statm] and
+  extracts the resident set size (RSS) in pages. It then converts this value to
+  megabytes by multiplying by the page size (4096 bytes) and dividing by 1024^2.
+  
+  @param pid The process ID to query memory usage for
+  @return [Some mb] where [mb] is the memory usage in megabytes if successful,
+      [None] if the process doesn't exist or if there's an error reading
+      the memory information *)
+let get_memory_usage_mb pid =
+  let filename = Printf.sprintf "/proc/%d/statm" pid in
+  try
+    let line = In_channel.read_all filename |> String.strip in
```

Comment:
I'd do this instead, it's clearer:
```ocaml
Scanf.sscanf line "%d %d %d %d %d %d %d"
    (fun size resident shared text lib data dt ->
       (size, resident, shared, text, lib, data, dt)
    )
```

In our case, we can just return `resident` instead of a tuple.